src/array.cr

```diff
@@ -1045,6 +1026,66 @@ class Array(T)
     self
   end
 
+  # Inserts all of the elements from *other* before the element at *index*.
+  #
+  # This method shifts the element currently at *index* (if any) and any
+  # subsequent elements to the right, increasing their indices. If the value
+  # of *index* is negative, counting starts from the end of the array.
```

Comment:
**suggestion:** Clarify the semantics of negative indices (because they differ from other methods):
```suggestion
  # of *index* is negative, counting starts from the end of the array.
  # For example, `-1` indicates insertion after the last element, `-2` before
  # the last element.
```